src/configurePipeline/services/source/sourceRepoService.ts

```diff
@@ -0,0 +1,172 @@
+import * as vscode from 'vscode';
+
+import * as fs from 'fs';
+import * as git from "simple-git/promise";
+import * as Mustache from "mustache";
+
+import { GitRepoDetails, SourceProviderType } from '../../model/Common';
+import { AzureDevOpsService } from "../azureDevOpsService";
+import Q = require('q');
+
+export class SourceRepoService {
+    private gitReference: git.SimpleGit;
+    constructor() {
+
+    }
+
+    /**
+     * @returns repoObject: localPath, provider, repoName, repoId, repoUrl, currentRemote, branch
+     */
+    public getSourceRepo(azureDevOps: AzureDevOpsService) {
+        // shows options:
+        // CurrentWorkspace
+        // Browse local folders
+        // Github: 1. Call github api (pat/authentication needed) 2. GetViaAzureDevops pipeline api (github service connection needed)
+        // AzureDevOps repo's: Call Azure DevOps Code api (credentials exist from azure-account extension)
+        let sourceOptions: string[] = ["Current Workspace", "Browse local machine", "Github repository"];
+        return vscode.window.showQuickPick(sourceOptions, {
+            placeHolder: "Select the folderor repository to deploy"
+        })
+            .then((selectedSourceOption) => {
+                let repoGitConfigPath = "";
+                switch (selectedSourceOption) {
+                    case sourceOptions[0]:
+                        repoGitConfigPath = vscode.workspace.workspaceFolders[0].uri.fsPath;
+                        break;
+                    case sourceOptions[1]:
+                    case sourceOptions[2]:
+                    default:
+                        repoGitConfigPath = vscode.workspace.workspaceFolders[0].uri.fsPath;
+                }
+                this.gitReference = git(repoGitConfigPath);
```

Comment:
i think below 2 lines are specific in case of local git repository and will not be required for GitHub. So it should be moved under specific case ?